modAionImpl/src/org/aion/zero/impl/db/AionRepositoryCache.java

```diff
@@ -309,6 +314,15 @@ public void saveCode(Address address, byte[] code) {
         return getContractDetails(address).getCode(codeHash);
     }
 
+    @Override
+    public byte[] getTransformedCode(Address address) {
```

Comment:
This method requires locking. I suggest using the full write locks to ensure correctness. We can improve performance later.

Reply:
for the getTransformedCode probably don't need locks? the Transformed code shouldn't be changed after deploy. Like the getcode method, they don't have locks also.

Reply:
to get the code we often need to load the accounts which requires write locks